extensions/apns_message_handler.go

```diff
@@ -223,45 +188,50 @@ func (a *APNSMessageHandler) HandleMessages(_ context.Context, message interface
 		"topic":     message.Topic,
 	})
 	l.Debug("received message to send to apns")
-	notification, err := a.buildNotification(message)
+
+	parsedNotification, err := a.parseKafkaMessage(message)
 	if err != nil {
+		l.WithError(err).Error("error parsing kafka message")
+		a.waitGroupDone()
+		apnsResMutex.Lock()
+		a.ignoredMessages++
+		apnsResMutex.Unlock()
```

Comment:
We don't need it and I really want to remove it. But it's used in several tests so it would require a rather big refactor so I did not do it now. We have a task for it, though.